tests2/gui/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+#  Copyright (C) 2023 KeePassXC Team <team@keepassxc.org>
+#
+#  This program is free software: you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation, either version 2 or (at your option)
+#  version 3 of the License.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+include_directories(
+        ${CMAKE_CURRENT_SOURCE_DIR}
+        ${CMAKE_CURRENT_BINARY_DIR})
+
+set(TEST_LIBRARIES keepassx_core)
+set(TEST_GUI_SUPPORT_SOURCES
+        main.cpp
+        ../../tests/util/TemporaryFile.cpp
+        FixtureBase.cpp
+        FixtureWithDb.cpp)
+add_library(TEST_GUI_SUPPORT STATIC ${TEST_GUI_SUPPORT_SOURCES})
+target_link_libraries(TEST_GUI_SUPPORT Qt5::Core Qt5::Widgets Qt5::Test Catch2::Catch2)
+
+# Add new unit tests here.
+# Naming convention: word-cased separated by dash.
+# Avoid naming with "test" as obvious and useless prefix.
+add_unit_test(NAME pass-phrase-generator-gui
+        SOURCES TestPassphraseGeneratorGui.cpp
+        LIBS TEST_GUI_SUPPORT ${TEST_LIBRARIES})
+
+add_unit_test(NAME totp-gui
+        SOURCES TestTotpGui.cpp
+        LIBS TEST_GUI_SUPPORT ${TEST_LIBRARIES})
```

Comment:
Now that we are using Catch2 we don't want to create a new test exe for every test. You want one TestGui.exe with multiple .cpp files and one main.cpp.

Reply:
I see, let me check it.

Reply:
I just fixed it :-)

Reply:
Thank you for fix. I think the same way we need to update the `CMakeLists.txt` for regular catch2 tests. I will do it...

About sub tests behaviour, according to [this tutorial](https://github.com/catchorg/Catch2/blob/devel/docs/tutorial.md#test-cases-and-sections):
> For each SECTION the TEST_CASE is executed from the start.

It is true for `SCENARIO` and nested `GIVEN/WHEN/THEN` as well.

I know it makes tests longer. One of possible workaround is to open the main window in the `main()` and pass somehow the created `m_mainWindow` into the tests. Let me play with it...

Reply:
I tried to find a way to pass an once created `m_mainWindow` into the tests but nothing found. The possible solution is to have a global shared variable. But do we really need it?
From other hand I compared the tests with and without DB - and sure thing with opening DB it starts much longer. But the same workflow we have in the current tests:
```c
// Every test starts with resetting config settings and opening the temp database
void TestGui::init()
```

The legacy single setup
```c
void TestGui::initTestCase()
{
    QVERIFY(Crypto::init());
    Config::createTempFileInstance();
    Application::bootstrap();

    m_mainWindow.reset(new MainWindow());
    m_tabWidget = m_mainWindow->findChild<DatabaseTabWidget*>("tabWidget");
    m_statusBarLabel = m_mainWindow->findChild<QLabel*>("statusBarLabel");
    m_mainWindow->show();
    m_mainWindow->resize(1024, 768);
}
```

is migrated to the `main()` and `FixtureBase` correspondingly:
```c
    Crypto::init();
    Config::createTempFileInstance();
    Application::bootstrap();
```

```c
    m_mainWindow.reset(new MainWindow());
    m_mainWindow->resize(1024, 768);
    m_mainWindow->show();
    m_mainWindow->activateWindow();
```
where the `FixtureBase` is executed pretty fast.
The slow execution is coming from the `FixtureWithDB` but as I wrote above we always reopen a DB for each test.

Reply:
FixtureWithDB should focus on the DatabaseWidget level of the GUI. Basically you pass in what database to open and it opens it and unlocks it then the test starts. Test ends with closing the database widget completely. The main window should remain static through all this.

Reply:
See my latest commit where I made the `m_mainWindow` as static field.
It works, saving couple seconds of 25s (tested on `TestPassphraseGeneratorGui.cpp`).
If the `static` field is fine let's do this way.

Reply:
Yah QPointer is correct.

Reply:
I noticed the latest commit broke the `TestPassphraseGeneratorGui` tests. It fails on
```c
REQUIRE(pPassPhraseWidget->isVisible());
```

Even with commented this assertion it fails on `"User presses Ctrl+R (regenerate)"` test:
```c
REQUIRE_FALSE(actualPhraseBefore == actualPhrase);
```

Will try to debug...

Reply:
I fixed the above issue by closing the widget after each test. Made as a separate commit.